### PR TITLE
fix: method should be updated when triggering

### LINKED ISSUE #615: Trigger disregards `setMethods` and remains referencing the old one
### Version
1.0.0-beta.16

### Reproduction link
[https://github.com/lchanmann/vue-test-utils-setMethods-issue](https://github.com/lchanmann/vue-test-utils-setMethods-issue)

### Steps to reproduce
Clone the repo, then run:

    yarn install
    yarn test

### What is expected?
Should pass tests

### What is actually happening?
Test fails

    FAIL  ./setMethods.test.js
     ● Console
   
       console.log setMethods.test.js:9
         clickMethod
   
     ● setMethods › calls stub when event is triggered
   
       expect(jest.fn()).toHaveBeenCalled()
   
       Expected mock function to have been called.
   
         at Object.<anonymous> (setMethods.test.js:17:29)
             at new Promise (<anonymous>)
             at <anonymous>
         at process._tickCallback (internal/process/next_tick.js:188:7)

**setMethods.test.js**

```javascript
import { mount } from '@vue/test-utils'

describe('setMethods', () => {
  it('calls stub when event is triggered', () => {
    const TestComponent = {
      template: '<button @click="clickMethod">Click here</button>',
      methods: {
        clickMethod() {
          console.log('clickMethod')
        }
      }
    }
    const wrapper = mount(TestComponent)
    const clickMethodStub = jest.fn()
    wrapper.setMethods({ clickMethod: clickMethodStub })
    wrapper.find('button').trigger('click')
    expect(clickMethodStub).toHaveBeenCalled()
  })
})
```

### PR DESCRIPTION
**Pull request by briwa**
Potentially fix #615.

I've learned about this bug a while ago before the issue was raised (it happened on my project as well). Here's a small isolated repro link https://jsfiddle.net/3vbnq15z/

I understand that `update` is deprecated and the test utils are going towards synchronous updates. However I think for `trigger` it is necessary to update. Let me know what you think about the approach.

I also deleted one test in `setMethod` spec which is a duplicate.

Thanks.

### CODE REVIEW

File: test/specs/wrapper/setMethods.spec.js [hunk 1]
```diff
@@ -10,18 +11,22 @@ describeWithShallowAndMount('setMethods', (mountingMethod) => {
...
   })
+
+  it('should replace methods when tied to an event', () => {
+    const wrapper = mountingMethod(ComponentWithEvents)
+    expect(wrapper.vm.isActive).to.be.false
+    wrapper.find('.toggle').trigger('click')
+    expect(wrapper.vm.isActive).to.be.true
+    // Replace the toggle function so that the data supposedly won't change
+    const toggleActive = () => console.log('overriden')
+    wrapper.setMethods({ toggleActive })
```
Comment (eddyerburgh) [inline test/specs/wrapper/setMethods.spec.js:29]: You should trigger toggle active again after you call setMethods, to make sure it's been overwritten

Comment (briwa) [reply test/specs/wrapper/setMethods.spec.js:29]: @eddyerburgh My bad, totally missed that part. I've updated the PR. Thanks.

**Review by eddyerburgh — APPROVED** [review]

**Comment by eddyerburgh** [comment]
Thanks :)
